From df2d704b310da4aff0592bb87ef68df4aca2240d Mon Sep 17 00:00:00 2001
From: Chenyan Feng <ella.feng@nxp.com>
Date: Tue, 24 Apr 2018 23:13:42 +0800
Subject: [PATCH 3625/5242] MGS-3832 [#imx-921] Fix the cl_convolution of ACL
 which cause MMU exception

commit  479f9f17b7e83c4c2a0ea12d76a4d74c139190ca from
https://source.codeaurora.org/external/imx/linux-imx.git

Dump the error info only when there is an error. If the gcdALLOC_ON_FAULT is not enabled, still need to check
the HW status to decide whether to dump the exception info or not

Signed-off-by: Ella Feng <ella.feng@nxp.com>
Reviewed-by: Xianzhong Li <xianzhong.li@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../mxc/gpu-viv/hal/kernel/gc_hal_kernel_event.c   |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mxc/gpu-viv/hal/kernel/gc_hal_kernel_event.c b/drivers/mxc/gpu-viv/hal/kernel/gc_hal_kernel_event.c
index 1246fe0..7a3e6ed 100644
--- a/drivers/mxc/gpu-viv/hal/kernel/gc_hal_kernel_event.c
+++ b/drivers/mxc/gpu-viv/hal/kernel/gc_hal_kernel_event.c
@@ -2215,8 +2215,8 @@
 #if gcdALLOC_ON_FAULT
             status = gckHARDWARE_HandleFault(Event->kernel->hardware);
 
-            if (gcmIS_ERROR(status))
 #endif
+            if (gcmIS_ERROR(status))
             {
                 /* Dump error is fault can't be handle. */
                 gckHARDWARE_DumpMMUException(Event->kernel->hardware);
-- 
1.7.9.5

